### PR TITLE
fix: [CI-6350]: "run in debug mode" execution action for CI pipelines

### PR DESCRIPTION
**Pull request by shaurya-harness**
### Summary

"run in debug mode" execution action for CI pipelines
#### Screenshots

Screenshot 2022-12-15 at 4 10 44 PM
Screenshot 2022-12-15 at 4 10 52 PM

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

**Comment by shaurya-harness** [comment]
retrigger jest

**Review by d1wilko — CHANGES_REQUESTED** [review]
small suggestions - but looking good

File: src/modules/70-pipeline/pages/execution-list/ExecutionList.tsx [hunk 1]
```diff
@@ -75,6 +75,7 @@ function ExecutionListInternal(props: ExecutionListProps): React.ReactElement {
 
   const { module } = useModuleInfo()
   const [viewCompiledYaml, setViewCompiledYaml] = React.useState<PipelineExecutionSummary | undefined>(undefined)
+  const [loadingonReRunInDebugMode, setLoadingOnReRunInDebugMode] = useState(false)
```
[the remark this answers is not in the document]
Comment (shaurya-harness) [reply src/modules/70-pipeline/pages/execution-list/ExecutionList.tsx:78]: sure

File: src/modules/70-pipeline/pages/execution-list/ExecutionListTable/ExecutionListCells.tsx [hunk 2]
```diff
@@ -247,12 +252,58 @@ export const DurationCell: CellType = ({ row }) => {
...
+    try {
+      const response = await runPipelineInDebugMode()
+      setLoadingOnReRunInDebugMode?.(true)
+      if (response.status === 'SUCCESS') {
+        showSuccess(getString('runPipelineForm.pipelineRunSuccessFully'))
+        setLoadingOnReRunInDebugMode?.(false)
+        history.push({
+          pathname: routes.toExecutionPipelineView({
+            orgIdentifier,
+            pipelineIdentifier: defaultTo(pipelineIdentifier || data.pipelineIdentifier, ''),
```
[the remark this answers is not in the document]
Comment (shaurya-harness) [reply src/modules/70-pipeline/pages/execution-list/ExecutionListTable/ExecutionListCells.tsx:293]: yes

File: src/modules/70-pipeline/pages/execution/ExecutionLandingPage/ExecutionLandingPage.tsx [hunk 3]
```diff
@@ -225,6 +229,48 @@ export default function ExecutionLandingPage(props: React.PropsWithChildren<unkn
...
+    try {
+      const response = await runPipelineInDebugMode()
+      if (response.status === 'SUCCESS') {
+        showSuccess(getString('runPipelineForm.pipelineRunSuccessFully'))
+        history.push({
+          pathname: routes.toExecutionPipelineView({
+            orgIdentifier,
+            pipelineIdentifier,
+            projectIdentifier,
+            executionIdentifier: defaultTo(response?.data?.planExecution?.uuid, ''),
```
[the remark this answers is not in the document]
Comment (shaurya-harness) [reply src/modules/70-pipeline/pages/execution/ExecutionLandingPage/ExecutionLandingPage.tsx:261]: yes

File: src/modules/70-pipeline/pages/execution-list/ExecutionListTable/ExecutionListCells.tsx [hunk 4]
```diff
@@ -247,12 +252,58 @@ export const DurationCell: CellType = ({ row }) => {
...
+  const { trackEvent } = useTelemetry()
+  const { mutate: runPipelineInDebugMode } = useDebugPipelineExecuteWithInputSetYaml({
+    queryParams: {
+      accountIdentifier: accountId,
+      projectIdentifier,
+      orgIdentifier,
+      moduleType: module || ''
+    },
+    identifier: defaultTo(pipelineIdentifier || data.pipelineIdentifier, ''),
+    originalExecutionId: defaultTo(data.planExecutionId, ''),
```
Comment (d1wilko) [inline src/modules/70-pipeline/pages/execution-list/ExecutionListTable/ExecutionListCells.tsx:275]: sorry can we change these 2 to ?? aswell - sorry I missed them

**Review by d1wilko — CHANGES_REQUESTED** [review]
sorry one last change

**Comment by shaurya-harness** [comment]
retrigger jest

File: src/modules/70-pipeline/pages/execution-list/ExecutionListTable/ExecutionListCells.tsx (hunk 4 above)
Comment (shaurya-harness) [reply src/modules/70-pipeline/pages/execution-list/ExecutionListTable/ExecutionListCells.tsx]: sure

**Review by d1wilko — APPROVED** [review]
LGTM :)

**Review by bhavya-sinha — APPROVED** [review]

**Review by ishantmehtaharness — APPROVED** [review]
Looks Good

**Review by vardanbansal-harness — APPROVED** [review]

**Pull request merged**